src/main/java/io/confluent/clientmetrics/MetricsPublisher.java

```diff
@@ -0,0 +1,106 @@
+package io.confluent.clientmetrics;
+
+import io.confluent.clientmetrics.avro.MetricNameAVRO;
+import io.confluent.clientmetrics.avro.MetricValueAVRO;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.StreamsConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class MetricsPublisher {
```

Comment:
I still don't see where this code came from?

Reply:
Done (there were not much code actually :) )

Reply:
```
  private void publishMetrics() {
    mySend(CLIENT_METRICS_TOPIC, "Some metrics...");
    // KStream topology
    // Some JMX metrics
  }
```